### PR TITLE
Improve Rules module documentation

### PR DESCRIPTION
**Pull request by agriffard**
The Rules reference was too limited to explain how rules are composed, evaluated, serialized, or extended. This update turns it into practical canonical guidance grounded in the current module, abstractions, Layers integration, and tests.

## Summary
- document activation, built-in conditions and operators, short-circuiting, scripting, recipes, deployment, permissions, and edge cases
- add compile-accurate examples for programmatic evaluation, custom conditions, display drivers, and operators
- add limitations and troubleshooting guidance for serialization, JavaScript, content types, and layer management

## Testing
- `python -m mkdocs build`
- compiled the documented C# examples against the current Orchard Core projects

### CODE REVIEW

File: src/docs/reference/modules/Rules/README.md [hunk 1]
```diff
@@ -1,23 +1,412 @@
...
-### Custom Conditions
+Rules are infrastructure. The module does not provide a standalone admin screen or decide what a successful rule does. A consuming feature stores a `Rule`, calls `IRuleService`, and supplies any authorization and management UI around it. The [Layers](../Layers/README.md) module is the built-in consumer: it uses rules to decide which widget layers are active for the current request.
 
-For more intricate scenarios, you have the option to craft your own conditions. To achieve this, you'll be required to implement the following abstractions from the `OrchardCore.Rules.Abstractions` package:
+## Enable the module
 
-- `Condition`
-- `ConditionEvaluator`
-- `ConditionDisplayDriver`
+Enable **Rules** from **Configuration** > **Features** in the admin. Enabling a feature that depends on `OrchardCore.Rules`, such as Layers, enables it automatically.
```
Comment (MikeAlhayek) [inline src/docs/reference/modules/Rules/README.md]:
```suggestion
Enable **Rules** from **Tool** > **Features** in the admin. Enabling a feature that depends on `OrchardCore.Rules`, such as Layers, enables it automatically.
```  

you should try using the skills from https://github.com/CrestApps/CrestApps.AgentSkills if you are using copilot or clause CLI, try the plugin will improve the outcome.

**Review by sebastienros — APPROVED** [review]
